src/sharepoint/rest/items.ts

```diff
@@ -1,42 +1,186 @@
 "use strict";
+
 import { Queryable } from "./Queryable";
 import * as Util from "../../utils/util";
 import * as Mixins from "./mixins";
+import { Gettable, SelectableGettable, FilterableSelectableGettable } from "./actionables";
+import { Folder } from "./folders";
+import { ContentType } from "./contenttypes";
+import { RoleAssignments } from "./roleassignments";
 
+/**
+ * Describes a collection of Item objects
+ * 
+ */
 export class Items extends Queryable implements Mixins.Gettable, Mixins.Selectable, Mixins.Filterable {
 
-    constructor(url: string | Queryable) {
-        super(url, "items");
+    /**
+     * Creates a new instance of the Items class
+     * 
+     * @param baseUrl The url or Queryable which forms the parent of this fields collection
+     */
+    constructor(baseUrl: string | Queryable) {
+        super(baseUrl, "items");
     }
 
+    /**
+     * Gets an Item by id
+     * 
+     * @param id The integer id of the item to retrieve
+     */
     public getById(id: number): Item {
         this.concat(`(${id})`);
```

Comment:
Unsure what you mean - there are many types of ids in SharePoint, GUID, integer and content type for example. In this case we are getting an item by id, which is expected to be an integer. What would you change?